### ISSUE TITLE
Support POST requests

### ISSUE THREAD

**Issue by porfirioribeiro** [porfirioribeiro/gatsby-plugin-netlify-functions#2]
Need to figure out a way to fix the POST requests not having the body parsed.

In #1 @willcosgrove tried to fix it, but it needs more investigation

**Comment by davetransom** [comment]
I'm not entirely sure of the specifics of the lambda implementation, but is this a case of just adding certain `body-parser` configuration?

```javascript
const bodyParser = require('body-parser')

exports.onCreateDevServer = (
  { app },
  { ... snip }
) => {
  app.use('/.netlify/functions/', bodyParser.text({ type: req => true }), (req, res, next) => {
    //... snip
  })
}
```

The example lambda functions I've seen seem to expect a text body (as opposed to parsing "application/json" or "application/x-www-form-urlencoded" automatically), then perform the parsing in the handler, like so:
```javascript
exports.handler = function(event, context, callback) {
  const data = JSON.parse(event.body)
  //  ... do something with data
}
```

Which seems to be backed up by https://www.netlify.com/docs/functions/, indicating the body should be a string in their example:
```javascript
{
  ...
  "body": "A JSON string of the request payload."
}
```

PS: The typo highlighted in #1 would still need to be fixed too I think i.e. `res.body` should be `req.body`.

EDIT: Oops. I didn't look at the correct branch on @willcosgrove's fork and missed the code for his actual changes. Sorry! His fork works well enough at least for the handling text (json) content.

**Comment by willcosgrove** [comment]
Yes you are correct, and that would probably cover most people’s use cases.  So that definitely seems like the right first step.  But to fix it comprehensively, we would also need to handle non text bodies, like image data or something.  It appears that in those cases, the lambda event should convert the body to a base64 encoding, and set a flag to indicate that the request body is base64 encoded.

I can’t find documentation on determining which requests will get base64 encoded.  I believe that on AWS, that setup is configured in the API Gateway service, and not in Lambda.  So it would depend on how Netlify has the API Gateway configured.

**Comment by porfirioribeiro** [comment]
I didn't had time yet to investigate this.
I'm open to PR's that help fixing this, and yes i think we can merge #1 as it is a typo, even if it does not fix this problem. 
If you want @willcosgrove re-open it and i'll merge it

**Comment by DavidSabine** [comment]
+1 upvote
